cookbooks/aws-parallelcluster-config/recipes/aws_cli.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: aws_cli
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Users to configure the AWS CLI for.
+# We configure the AWS CLI for all the users that are created by ParallelCluster and must interact with AWS.
+users = node["cluster"]["head_node_imds_allowed_users"]
+
+# AWS CLI configurations to set.
+# Note: only the configurations with a not empty value will be set.
+region = node["cluster"]["region"]
+configurations = {
+  "ca_bundle" =>  region.start_with?("us-iso") ? "/etc/pki/#{region}/certs/ca-bundle.pem" : "",
+}
+
+def set_aws_cli_configuration(config_name, expected_value, user)
+  execute "Setting AWS CLI configuration for user #{user}: #{config_name} set to '#{expected_value}'" do
+    user user
+    login true
+    command "aws configure set #{config_name} #{expected_value}"
+  end
+end
+
+users.each do |user|
+  configurations.each do |config_name, config_value|
+    next if config_value.nil? || config_value.empty?
+    set_aws_cli_configuration(config_name, config_value, user)
```

Comment:
Actually it does only for ISO regions.
The block is able to apply an AWS CLI configuration regardless the region, but it is applied only if you provide values to set and such values are set only for ISO regions.

Reply:
sorry for the delayed response, I was actually thinking we wouldn't want this object `ca-bundle` to exist at all for non adc regions since we didn't need it before.